Add private beta LLM Gateway docs [closes ENT-1059] (#3999)
langchain-ai/docs: src/langsmith/llm-gateway-access.mdx

---
title: Traces, Engine, and access control
description: Understand where gateway traces land, how policy violations surface in LangSmith Engine, and who can see and configure what.
hidden: true
---

<Note>
**Private beta:** The LLM Gateway is in private beta. [Sign up for the waitlist](https://www.langchain.com/langsmith-llm-gateway-waitlist) to get access.
</Note>

Every call through the LLM Gateway is traced to LangSmith, and policy violations surface in [LangSmith Engine](/langsmith/engine) for triage.

## Where gateway traces appear

### Default: the gateway project

By default, all gateway-proxied calls are traced to a project named `gateway` in the [workspace](/langsmith/administration-overview#workspaces) associated with the caller's API key. This project is accessible to [Organization admins](/langsmith/rbac/#organization-admin) with gateway edit permissions.

### Project routing

Gateway-proxied LLM calls always land in a fixed `gateway` project, as well as a API-key specific project with the scheme `gateway-<short_api_key>-<api_key_id>`.

> Set [ABAC](/langsmith/abac) policies against these individual projects to limit view access to only the intended audience.

Reviewer: ```suggestion
By default, all gateway-proxied calls are traced to a project named `gateway` in the [workspace](/langsmith/administration-overview#workspaces) associated with the caller's API key, as well as a API-key specific project with the scheme `gateway-<short_api_key>-<api_key_id>`.

Control access to these tracing projects with [RBAC](/langsmith/rbac) and [ABAC](/langsmith/abac)
```

> Gateway-proxied LLM calls always land in a fixed `gateway` project. The gateway sets the `Langsmith-Project` header itself on the outbound OTLP upload. It does **not** read a `Langsmith-Project` header (or SDK-set project) from the caller, so traces cannot be routed into a caller's own project (for example, `my-agent`). For per-API-key isolation in the UI, each trace is also copied to a second project named `gateway-<short_api_key>-<api_key_id>`. Gateway-specific metadata is attached to the spans in both copies.

Reviewer: ```suggestion
Gateway-proxied LLM calls always land in a fixed `gateway` project, as well as a API-key specific project with the scheme `gateway-<short_api_key>-<api_key_id>`.
```

- **Cost data:** token counts and cost are computed inline and feed the same spend accumulator that spend-cap policies enforce against.

## LangSmith Engine integration

When a governance policy fires (such as when a spend limit is hit, PII is detected and redacted, or a secret is caught), the event is recorded as metadata on the trace. These policy violations surface as issues in LangSmith Engine.

From an Engine issue, you can:

1. **See the violation:** which policy fired, what was blocked or redacted.
1. **Click through to the trace:** see exactly what the agent was doing when the policy triggered.
1. **Diagnose the root cause:** was it a retry loop burning budget, a user pasting credentials into a prompt, or a legitimate workload that outgrew its cap, etc.
1. **Take action:** update the agent's configuration, adjust the policy, or escalate.

## Audit logging

The gateway logs two categories of events:

| Category | What's logged |
| --- | --- |
| **Administrative changes** | Policy creation, modification, and deletion. Role and permission changes related to gateway access. |

> | **Gateway invocations** | Each proxied call, including the caller identity, provider, model, and policy evaluation result. |

Reviewer: ```suggestion
| **Gateway invocations** | Each proxied call, including the caller identity and matched policy IDs . |
```

[Audit logs](/langsmith/audit-logs) are available to organization admins on the [Enterprise plan](/langsmith/pricing-plans).

## Permissions

### Required permissions

| Action | Permission needed | Who has it by default |
| --- | --- | --- |
| Make calls through the gateway | `gateway:invoke` + `workspaces:read` | `WORKSPACE_ADMIN` only |
| Create, edit, or delete policies | `organization:manage` | Org admins |
| View gateway traces | `projects:read` + `runs:read` | `WORKSPACE_ADMIN`, `WORKSPACE_USER`, `WORKSPACE_VIEWER` |
| View audit logs | `organization:manage` | Org admins |

The built-in `WORKSPACE_USER` and `WORKSPACE_VIEWER` roles do **not** include `gateway:invoke` and cannot be edited. To grant gateway access without full workspace-admin privileges, create a custom workspace role with `gateway:invoke` and `workspaces:read` (requires an RBAC-enabled plan). For instructions, refer to [Admin setup](/langsmith/llm-gateway-admin-setup).

### API key scoping

Always use workspace-scoped API keys for the gateway. Organization-scoped keys are not supported for invoking the gateway.

> Each API key can carry its own spend limit and policy bindings to scope agents independently.

Reviewer: ```suggestion
```

> Always use workspace-scoped API keys for the gateway. Organization-scoped keys are not recommended because the gateway's permission checks and trace routing are both workspace-scoped. A workspace-scoped key gives you clean policy matching and ensures traces land in the right project.

Reviewer: ```suggestion
Always use workspace-scoped API keys for the gateway. Organization-scoped keys are not supported for invoking the gateway.
```

For this to work as intended, don't distribute provider API keys to developers alongside gateway access. The gateway centralizes provider credentials by default. A pass-through mode also exists for OAuth-based flows like Claude Code Max—contact us if your organization wants to allow or restrict it.

### Restricting trace visibility

Gateway traces are written as runs into a workspace project and follow LangSmith's standard workspace membership model. Anyone with `runs:read` (and `projects:read` to see the project itself) on the workspace can view traces in that workspace's gateway project. The built-in roles `WORKSPACE_ADMIN`, `WORKSPACE_USER`, and `WORKSPACE_VIEWER` all include both permissions by default.

If you need to limit who can see gateway traces, you have two options:

- **Separate workspaces** (works on any [plan](/langsmith/pricing-plans)): create one workspace with restricted membership and another for developer coding agents with broader membership. Each workspace has its own provider secrets and trace projects.

> - **Project-level access policies** (requires an [Enterprise plan](/langsmith/pricing-plans)): write a policy restricting `projects:read` and `runs:read` on the gateway project to specific users or roles.

Reviewer: ```suggestion
- **Project-level access policies** (requires an [Enterprise plan](/langsmith/pricing-plans)): write an ABAC policy restricting `projects:read` and `runs:read` on the gateway project to specific users or roles.
```
